payroll/models/hr_payslip.py

```diff
@@ -440,35 +509,26 @@ def _sum_salary_rule_category(localdict, category, amount):
 
             return localdict
 
+        payslip = self.env["hr.payslip"].browse(payslip_id)
+        contracts = self.env["hr.contract"].browse(contract_ids)
+
         # we keep a dict with the result because a value can be overwritten by
         # another rule with the same code
         result_dict = {}
         rules_dict = {}
-        worked_days_dict = {}
-        inputs_dict = {}
+        contract_dict = {}
         blacklist = []
-        payslip = self.env["hr.payslip"].browse(payslip_id)
-        for worked_days_line in payslip.worked_days_line_ids:
-            worked_days_dict[worked_days_line.code] = worked_days_line
-        for input_line in payslip.input_line_ids:
-            inputs_dict[input_line.code] = input_line
-
         categories = BrowsableObject(payslip.employee_id.id, {}, self.env)
-        inputs = InputLine(payslip.employee_id.id, inputs_dict, self.env)
-        worked_days = WorkedDays(payslip.employee_id.id, worked_days_dict, self.env)
-        payslips = Payslips(payslip.employee_id.id, payslip, self.env)
         rules = BrowsableObject(payslip.employee_id.id, rules_dict, self.env)
+        this_contract = BrowsableObject(payslip.employee_id.id, contract_dict, self.env)
```

Comment:
I suggest changing this to "contract" or something more generic because "this_contract" will be easily confused.

Reply:
current_contract sounds fine.